system-test/bigtable.js

```diff
@@ -0,0 +1,732 @@
+                });
+              });
+            });
+
+            done();
+          });
+        });
+
+        it('should run a regex against the row id', function(done) {
+          var filter = {
+            row: /[a-z]+on$/
+          };
+
+          TABLE.getRows({ filter: filter }, function(err, rows) {
+            assert.ifError(err);
+
+            var keys = rows.map(function(row) {
+              return row.id;
+            }).sort();
+
+            assert.deepEqual(keys, [
+              'gwashington',
+              'tjefferson'
+            ]);
+
+            done();
+          });
+        });
+
+        it('should run a sink filter', function(done) {
+          var filter = [{
+            row: 'alincoln'
+          }, {
+            family: 'follows'
+          }, {
+            interleave: [
+              [{
+                all: true
+              }], [{
+                label: 'prezzy'
+              }, {
+                sink: true
+              }]
+            ]
+          }, {
+            column: 'gwashington'
+          }];
+
+          TABLE.getRows({ filter: filter }, function(err, rows) {
+            assert.ifError(err);
+
+            var columns = Object.keys(rows[0].data.follows).sort();
+
+            assert.deepEqual(columns, [
+              'gwashington',
+              'jadams',
+              'tjefferson'
+            ]);
+
+            done();
+          });
+        });
+
+        it('should accept a date range', function(done) {
+          var filter = {
+            time: {
+              start: new Date('March 21, 1986'),
+              end: new Date('March 23, 1986')
+            }
+          };
+
+          TABLE.getRows({ filter: filter }, function(err, rows) {
+            assert.ifError(err);
+            assert(rows.length > 0);
+            done();
+          });
+        });
+
+      });
+
+    });
+
+    describe('deleting rows', function() {
+
+      it('should delete specific cells', function(done) {
+        var row = TABLE.row('alincoln');
+
+        row.deleteCells(['follows:gwashington'], function(err) {
+          assert.ifError(err);
+          done();
+        });
+      });
+
+      it('should delete a family', function(done) {
+        var row = TABLE.row('gwashington');
+
+        row.deleteCells(['traits'], function(err) {
+          assert.ifError(err);
+          done();
+        });
+      });
+
+      it('should delete all the cells', function(done) {
+        var row = TABLE.row('alincoln');
+
+        row.delete(function(err) {
+          assert.ifError(err);
+          done();
+        });
+      });
+
+      it('should delete all the rows', function(done) {
+        TABLE.deleteRows(function(err) {
+          assert.ifError(err);
+
+          TABLE.getRows()
+            .on('error', done)
+            .on('data', function() {
+              done(new Error('Row found!'));
+            })
+            .on('end', done);
```

Comment:
It might be easier to follow with just:

```js
Table.getRows(function(err, rows) {
  assert.ifError(err);
  assert.strictEqual(rows.length, 0); // All rows were deleted
  done();
});
```